app/db/PostgresResults.java

```diff
@@ -15,136 +15,194 @@
 package db;
 
 import exceptions.GroundDbException;
-import exceptions.GroundException;
-
-import java.sql.ResultSet;
-import java.sql.SQLException;
-
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.Iterator;
 
-public class PostgresResults {
+/**
+ * A wrapper for Postgres's result queries.
+ *
+ * WARNING: Due to how Java implements ResultSet, an instance of this
+ * class can only be iterated through once. In addition, columns must be
+ * queried in the same order they exist in the database.
+ */
+public class PostgresResults implements DbResults {
   private static final Logger LOGGER = LoggerFactory.getLogger(PostgresResults.class);
-
   private final ResultSet resultSet;
+  private final boolean isEmpty;
 
   public PostgresResults(ResultSet resultSet) {
     this.resultSet = resultSet;
-  }
 
-  /**
-   * Move on to the next column in the result set.
-   *
-   * @return false if there are no more rows
-   */
-  public boolean next() throws GroundDbException {
+    boolean isEmpty = true;
     try {
-      return this.resultSet.next();
+      // Move cursor to row 1.
+      isEmpty = !this.resultSet.next();
     } catch (SQLException e) {
-      throw new GroundDbException(e);
```

Comment:
I wasn't sure if constructors should throw an exception; I'll see if I can do that instead.